services/checks/funcs.go

```diff
@@ -80,3 +85,80 @@ func formatVersionNum(args ...interface{}) (interface{}, error) {
 	}
 	return p.String(), nil
 }
+
+// GetAdditionalContext returns additional functions to be used in check scripts.
+func GetAdditionalContext() map[string]starlark.GoFunc {
+	return map[string]starlark.GoFunc{
+		"ip_is_private":      ipIsPrivate,
+		"parse_version":      parseVersion,
+		"format_version_num": formatVersionNum,
+	}
+}
+
+// ipIsPrivate accepts a single string argument (IP address) and
+// returns true for a private address, otherwise false.
+func ipIsPrivate(args ...interface{}) (interface{}, error) {
+	if l := len(args); l != 1 {
+		return nil, errors.Errorf("expected 1 argument, got %d", l)
+	}
+
+	ip, ok := args[0].(string)
+	if !ok {
+		return nil, errors.Errorf("expected string argument, got %[1]T (%[1]v)", args[0])
+	}
+
+	ipAddress := net.ParseIP(ip)
+	if ipAddress == nil {
+		// check if string was in CIDR notation
+		_, net, err := net.ParseCIDR(ip)
+		if err != nil {
+			return nil, errors.Errorf("invalid ip address: %s", ip)
+		}
+		for _, network := range privateNetworks {
+			if reflect.DeepEqual(net, network) {
```

Comment:
I'm pretty sure that DeepEqual will not work there. For example, both `10.0.0.0/8` and `10.0.0.0/9` are not public, because `10.0.0.0/8` is a superset of `10.0.0.0/9`

Reply:
you were right, changed the logic there and also changed a test case to check for `10.0.0.0/9` returns true now